libraries/AP_Scripting/examples/Serial_Dump.lua

```diff
@@ -0,0 +1,44 @@
+-- this script reads data from a serial port and dumps it to a file
+local file_name = 'serial dump.txt'
+local baud_rate = 9600
+
+-- find the serial first (0) scripting serial port instance
+local port = serial:find_serial(0) -- SERIALx_PROTOCOL 28
+
+if not port then
+    error("Could not find Scripting Serial Port")
+    return
+end
```

Comment:
```suggestion
local port = assert(serial:find_serial(0),"Could not find Scripting Serial Port")  -- SERIALx_PROTOCOL 28
```
I regret not pointing more people at the assert helper originally, so it's worth pointing out here, as it's a much cleaner snippet of code. I think we should be trying to encourage assert as much as possible here moving forward, for more details on this PIL has a good section [here](https://www.lua.org/pil/8.3.html)

Reply:
fixed